test/nodes/test_shaper.py

```diff
@@ -561,3 +577,54 @@ def test_basic_function_batch_invocation(tmp_path):
     assert "queries" in result
     assert len(result["meta"]) > 0
     assert "questions" in result["meta"][0]["invocation_context"]
+
+
+def test_prompt_node_with_shaper(tmp_path):
+    # test that the prompt node works with the shaper node
+    # here we use shaper to expand the query to the size of documents and rename the output to questions
+    with open(tmp_path / "tmp_config.yml", "w") as tmp_file:
+        tmp_file.write(
+            f"""
+            version: ignore
+            components:
+              - name: pmodel
+                type: PromptModel
+              - name: shaper
+                params:
+                  inputs:
+                    query:
```

Comment:
Since we only specify how to handle shaping the `query` input, does this mean the `documents` that is provided through the call below
```python
   result = pipeline.run(
        query="What's Berlin like?",
        documents=[Document("Berlin is an amazing city."), Document("Berlin is a cool city in Germany.")],
    )
```
is passed on as is to the `PromptNode`? If so I think we should make it clear in the docs of the `Shaper` node will ignore and pass on all variables not defined in its `inputs:`. What do you think?

Reply:
Yes and yes. Ok, I thought this was obvious, but yes, it is worth a mention for sure.

Reply:
Done, see https://github.com/deepset-ai/haystack/pull/3815/commits/bad46b53e40257d5c8655aa046b9f7b5b558ddcb